src/core/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,335 @@
+      </div>
+      {children}
+      <div className='iui-carousel-navigation-right'>
+        <IconButton
+          styleType='borderless'
+          size='small'
+          tabIndex={-1}
+          onClick={() =>
+            setCurrentIndex((old) => (slideCount + old + 1) % slideCount)
+          }
+          data-pressed={keysPressed['ArrowRight'] || undefined}
+        >
+          <SvgChevronRight />
+        </IconButton>
+      </div>
+    </nav>
+  );
+};
+
+const CarouselDots = (
+  props: {
+    length?: number;
+    maxCount?: number;
+    currentIndex?: number;
+    onSlideChange?: (index: number) => void;
+  } & React.ComponentPropsWithoutRef<'div'>,
+) => {
+  const {
+    maxCount = 5,
+    currentIndex: userCurrentIndex,
+    length,
+    className,
+    onSlideChange,
+    children,
+    ...rest
+  } = props;
+
+  useTheme();
+
+  const context = React.useContext(CarouselContext);
+  const slideCount = length ?? context?.slideCount;
+  const currentIndex = userCurrentIndex ?? context?.currentIndex;
+  const idPrefix = props.id ?? context?.idPrefix;
+
+  const handleSlideChange = React.useCallback(
+    (index: number) => {
+      context?.setCurrentIndex(index);
+      onSlideChange?.(index);
+    },
+    [context, onSlideChange],
+  );
+
+  if (
+    slideCount == undefined ||
+    currentIndex == undefined ||
+    (!idPrefix && !children)
+  ) {
+    throw new Error(
+      'CarouselDots needs to be used inside Carousel, or missing props need to be specified',
+    );
+  }
+
+  const dots = React.useMemo(
+    () =>
+      Array(slideCount)
+        .fill(null)
+        .map((_, index) => {
+          const firstDotIndex = Math.max(
+            0,
+            Math.min(
+              currentIndex - Math.ceil(maxCount / 2) + 1,
+              slideCount - maxCount,
+            ),
+          );
+          const lastDotIndex = Math.min(
+            slideCount - 1,
+            Math.max(currentIndex + Math.floor(maxCount / 2), maxCount - 1),
+          );
+
+          const isFirstSmallDot =
+            (index === firstDotIndex && index !== 0) ||
+            (index === lastDotIndex && index !== slideCount - 1);
+          const isSecondSmallDot =
+            (index === firstDotIndex + 1 && index !== 1) ||
+            (index === lastDotIndex - 1 && index !== slideCount - 2);
+
+          return (
+            <CarouselDot
+              aria-label={`Slide ${index}`}
+              isActive={index === currentIndex}
+              onClick={() => handleSlideChange(index)}
+              isFirst={isFirstSmallDot}
+              isSecond={isSecondSmallDot}
+              isInvisible={index < firstDotIndex || index > lastDotIndex}
+              key={index}
+              id={`${idPrefix}--dot-${index}`}
+              aria-controls={`${idPrefix}--slide-${index}`}
+            />
+          );
+        }),
+    [slideCount, currentIndex, maxCount, idPrefix, handleSlideChange],
+  );
+
+  return (
+    <div
```

Comment:
Shouldn't this be `ol` or `ul` and items `li` elements?

Reply:
I changed it because the `tab` and `tablist` roles should be enough and it was adding an extra element for no reason. Current approach will reduce the chance of misuse (we are expecting dots to be usable outside carousel).

W3 has the same in their [example](https://www.w3.org/TR/wai-aria-practices-1.2/examples/carousel/carousel-2-tablist.html) 
![image](https://user-images.githubusercontent.com/9084735/156778805-4805799f-031f-4834-8c9d-be2f1b417e88.png)